internal/venice-common/src/main/proto/VeniceReadService.proto

```diff
@@ -0,0 +1,27 @@
+syntax = 'proto3';
+package com.linkedin.venice.protocols;
+
+option java_multiple_files = true;
+
+service VeniceReadService {
+  rpc get (VeniceClientRequest) returns (VeniceServerResponse) {}
+  rpc batchGet(VeniceClientRequest) returns (VeniceServerResponse) {}
+}
+
+message VeniceClientRequest {
+  uint32 partition = 1;
+  string keyString = 2;           // used for single get
+  bytes  keyBytes = 3;            // used for batch get
```

Comment:
Can we reuse one field: `keyBytes` for both request types?

Reply:
we can, but then we would have to obtain the `keyBytes` on the client side. As of today, when the server receives a `single-get` request, `GetRouterRequest.parseGetHttpRequest` constructs a `GetRouterRequest` object to pass to `StorageReadRequestsHandler`. Here, `GetRouterRequest` has the key in the form of a string, where the method `getKeyBytesFromUrlKeyString` will decode the query string to get corresponding bytes for the key. We would have to perform this step on the client side if we were to reuse the same field for both request types.